rust/sealevel/programs/interchain-accounts/src/processor.rs

```diff
@@ -0,0 +1,507 @@
+
+    storage.ism = ism;
+
+    // Store it
+    InterchainAccountStorageAccount::from(storage).store(storage_info, false)?;
+
+    Ok(())
+}
+
+/// Accounts:
+/// 0. `[]` Storage PDA account.
+fn get_interchain_security_module(program_id: &Pubkey, accounts: &[AccountInfo]) -> ProgramResult {
+    let accounts_iter = &mut accounts.iter();
+
+    // Account 0: Storage PDA account.
+    let storage_info = next_account_info(accounts_iter)?;
+    let (expected_storage_pda_key, _expected_storage_pda_bump) =
+        Pubkey::find_program_address(program_storage_pda_seeds!(), program_id);
+    if storage_info.key != &expected_storage_pda_key {
+        return Err(ProgramError::InvalidArgument);
+    }
+    let storage =
+        InterchainAccountStorageAccount::fetch(&mut &storage_info.data.borrow()[..])?.into_inner();
+
+    set_return_data(
+        &storage
+            .ism
+            .try_to_vec()
+            .map_err(|err| ProgramError::BorshIoError(err.to_string()))?[..],
+    );
+
+    Ok(())
+}
+
+fn set_account_meta_return_data(program_id: &Pubkey) -> ProgramResult {
+    let (storage_pda_key, _storage_pda_bump) =
+        Pubkey::find_program_address(program_storage_pda_seeds!(), program_id);
+
+    let account_metas: Vec<SerializableAccountMeta> =
+        vec![AccountMeta::new(storage_pda_key, false).into()];
+
+    // Wrap it in the SimulationReturnData because serialized account_metas
+    // may end with zero byte(s), which are incorrectly truncated as
+    // simulated transaction return data.
+    // See `SimulationReturnData` for details.
+    let bytes = SimulationReturnData::new(account_metas)
+        .try_to_vec()
+        .map_err(|err| ProgramError::BorshIoError(err.to_string()))?;
+    set_return_data(&bytes[..]);
+
+    Ok(())
+}
+
+/// Enrolls remote routers.
+///
+/// Accounts:
+/// 0. `[executable]` System program.
+/// 1. `[writeable]` Storage PDA account.
+/// 2. `[signer]` Owner.
+fn enroll_remote_routers(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    configs: Vec<RemoteRouterConfig>,
+) -> ProgramResult {
+    let accounts_iter = &mut accounts.iter();
+
+    // Account 0: System program.
+    let system_program_info = next_account_info(accounts_iter)?;
+    if system_program_info.key != &system_program::id() {
+        return Err(ProgramError::InvalidArgument);
+    }
+
+    // Account 1: Storage PDA account.
+    let storage_info = next_account_info(accounts_iter)?;
+    let (expected_storage_pda_key, _expected_storage_pda_bump) =
+        Pubkey::find_program_address(program_storage_pda_seeds!(), program_id);
+    if storage_info.key != &expected_storage_pda_key {
+        return Err(ProgramError::InvalidArgument);
+    }
+    let mut storage =
+        InterchainAccountStorageAccount::fetch(&mut &storage_info.data.borrow()[..])?.into_inner();
+
+    // Account 2: Owner.
+    let owner_info = next_account_info(accounts_iter)?;
+    storage.ensure_owner_signer(owner_info)?;
```

Comment:
nit: I believe `enroll_remote_routers_only_owner` does this `ensure_owner_signer` as part of its logic, so no need to have this twice